api/api_public_blockchain.go

```diff
@@ -380,28 +381,53 @@ func (s *PublicBlockChainAPI) DoEstimateGas(ctx context.Context, b Backend, args
 	cap = hi
 
 	// Create a helper to check if a gas allowance results in an executable transaction
-	executable := func(gas uint64) bool {
+	executable := func(gas uint64) (bool, []byte, error, error) {
 		args.Gas = hexutil.Uint64(gas)
-
-		_, _, _, failed, err := DoCall(ctx, b, args, rpc.NewBlockNumberOrHashWithNumber(rpc.LatestBlockNumber), vm.Config{UseOpcodeComputationCost: true}, localTxExecutionTime, gasCap)
-		if err != nil || failed {
-			return false
+		ret, _, _, status, err := DoCall(ctx, b, args, rpc.NewBlockNumberOrHashWithNumber(rpc.LatestBlockNumber), vm.Config{}, 0, gasCap)
+		if err != nil {
+			if errors.Is(err, blockchain.ErrIntrinsicGas) {
+				// Special case, raise gas limit
+				return false, ret, nil, nil
+			}
```

Comment:
It seems a key change for the `eth_call` in this PR. If this is correct, the unnecessary changes (e.g., changed function signature, etc)  would not be made.

Reply:
I put it in in advance before appling the dynamic gas fee.